File: jiaoji_app/src/Component/ActivityDetails.js
import React from "react";
import { Card as AntCard } from "antd";
import "../css/ActivityDetail.css";

const ActivityDetails = ({ activity }) => {
    return (
        <div className="card-container">
            <AntCard className="activity-card ant-card-hoverable">
                <div className="ant-card-head">
                    <h2 className="ant-card-head-title">{activity.name}</h2>
                </div>
                <div className="ant-card-body">
                    <p>{activity.content}</p>
                    <ul>
                        <li>地点：{activity.location}</li>
                        <li>报名时间：{activity.signupTime}</li>
                        <li>活动时间：{activity.activityTime}</li>
                        <li>开设学院：{activity.departments}</li>
                        <li>报名限制：{activity.signupRestriction}</li>
                        <li>招募人数：{activity.recruitmentNumber}</li>
                        <li>主办单位：{activity.organizer}</li>
                        <li>素拓奖励：{activity.suScore} 分</li>
                        <li>劳动教育学时：{activity.laborHour} 时</li>
                    </ul>
                </div>
            </AntCard>
        </div>
    );
};

export default ActivityDetails;

File: jiaoji_app_back/jiaoji_app_back/src/main/java/com/example/jiaoji_app_back/entity/ActivityDetails.java
/*author: qyl*/
package com.example.jiaoji_app_back.entity;

import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Data
@NoArgsConstructor

@Table(name ="activity_details")
public class ActivityDetails {
    public enum Status {
        REJECTED,
        NOT_RELEASE,
        TODO,
        PASS,
        SIGN,
        PROCESS,
        OVER;
    }


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String content;
    private String location;
    private String signupTime;
    private String activityTime;
    private String departments;
    private String signupRestriction;
    private String college;
    private Integer grade;
    private String club;
    private Long recruitmentNumber;
    private Long remainingNumber;
    private String organizer;
    private Long suScore;
    private Long laborHour;
    private Integer status;
    private String comments;
    private String photo;
    private String type;

    public ActivityDetails(Long id, String name, String content, String location, String signupTime, String activityTime, String departments, String signupRestriction, String college, Integer  grade, String club, Long recruitmentNumber, Long remainingNumber,String organizer, Long suScore, Long laborHour, Integer status, String comments, String photo, String type) {
        this.id = id;
        this.name = name;
        this.content = content;
        this.location = location;
        this.signupTime = signupTime;
        this.activityTime = activityTime;
        this.departments = departments;
        this.signupRestriction = signupRestriction;
        this.college = college;
        this.grade = grade;
        this.club = club;
        this.recruitmentNumber = recruitmentNumber;
        this.remainingNumber = remainingNumber;
        this.organizer = organizer;
        this.suScore = suScore;
        this.laborHour = laborHour;
        this.status = status;
        this.comments = comments;
        this.photo = photo;
        this.type = type;
    }


}